test/jdk/java/awt/Window/MultiWindowApp/ChildAlwaysOnTopTest.java

```diff
@@ -244,6 +220,32 @@ public void run() {
             }
         }
     }
+    // to check if the current top window background color
+    // matches the expected color
+    private static void checkTopWindow(int caseNo, int scenarioNo,
+                                       Color expectedColor)
+            throws InterruptedException, InvocationTargetException {
+
+        robot.delay(500);
+        robot.waitForIdle();
+        Color actualColor = robot.getPixelColor(point.x + 100, point.y + 100);
+
+        saveScreenCapture(caseNo , scenarioNo);
+
+        if (!actualColor.equals(expectedColor)) {
+            errorFlag = true;
+            SwingUtilities.invokeAndWait(()-> {
+                System.out.println("Green Window Active?: "+ win1.isActive());
+                System.out.println("Red Window Active? "+ win2.isActive());
+            });
+            System.out.println("Case "+ caseNo + errorMsg[scenarioNo - 1]
+                    +" Expected Color: "+ expectedColor +" vs Actual Color: "
+                    + actualColor);
```

Comment:
Here instead of just printing out the error i would save it into the static StringBuilder that i would allocate at the beginning of the test. Then when we are concluding testing i would just check if the buffer is empty and if it is not i would throw an RuntimeException with the buffer content. This way we would have all the errors reported by the JavaTest suite in the HTML report and this will make it into the whatever testing environment report we are running tests in. Also that would replace the errorFlag variable since emptiness of the buffer will be an indication of test pass.